### ISSUE TITLE
Clarify the process for the new-rp-namespace labels in the specs repo

### ISSUE THREAD

**Issue by maririos** [Azure/azure-sdk-tools#8362]
With [PR](https://devdiv.visualstudio.com/DevDiv/_git/openapi-alps/pullrequest/552463) the labels `new-rp-namespace-Approved` and `new-rp-namespace` were introduced.

The guidance for the user currently says: "once reviewed and approved by the Azure SDK team, ensure this PR has `new-rp-namespace-Approved` label". The wording could be changed to say `provide the link of the GitHub issue with the namespace approval and add the label ...` .

There has been confusion from the ARM reviewers and from the users on what to do in PRs when the label is requested.

Also, I am curious how this process started? Eng experience wasn't notified and neither the PM that manages the process @ronniegeraghty .

FYI @ladonnaq

**Comment by konrad-jamrozik** [comment]
Hey, sorry, I caused some confusion by adding this label recently. Previously, before my recent changes, the source said the following in [comment.yml](https://github.com/Azure/azure-rest-api-specs/blob/95381430ee109af5d02330184c2dc69f8ab31d38/.github/comment.yml):

``` yaml
- rule:
    type: label
    label: new-rp-namespace
    onLabeledComments: "Hi, @${PRAuthor}, our workflow has detected that there is no management SDK ever released for your RP, to further process SDK onboard for your RP, you should have the SDK client library name of your RP reviewed and approved. </br> <b>Action Required</b>: <li> Follow this guidance [Naming for new initial management or client libraries (new SDKs) - Overview (azure.com)](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/821/Naming-for-new-initial-management-or-client-libraries-(new-SDKs)) to create an issue for management client library name arch board review. </li> <li>Paste the issue link that you created in step 1 in this PR</li> </br> <b> Impact</b>: SDK release owner will take the approved management client library name to release SDK. No client library name approval will leads to SDK release delayed."
```

Here it is pretty-printed:

> Hi, @${PRAuthor}, our workflow has detected that there is no management SDK ever released for your RP, to further process SDK onboard for your RP, you should have the SDK client library name of your RP reviewed and approved. 
> 
> **Action Required**:
> 
> Follow this guidance [Naming for new initial management or client libraries (new SDKs) - Overview (azure.com)](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/821/Naming-for-new-initial-management-or-client-libraries-(new-SDKs)) to create an issue for management client library name arch board review. 
> 
> Paste the issue link that you created in step 1 in this PR
> 
> **Impact**: SDK release owner will take the approved management client library name to release SDK. No client library name approval will leads to SDK release delayed.


I have migrated this logic to `openapi-alps` in this PR, also adding the `new-rp-namespace-Approved` label requirement:

[Pull Request 552463](https://devdiv.visualstudio.com/DevDiv/_git/openapi-alps/pullrequest/552463): In preparation for `workflow-bot` deletion: migrate rules `Approved-OkToMerge`, `ArcReview` and `new-rp-namespace`

**Comment by ladonnaq** [comment]
@ronniegeraghty - please review additional details provided by Konrad. Also, consider how to handle the scenario when a new service/product lifecycle is private preview and the naming review has not concluded.  Teams will be blocked from merging their private preview API spec into RPSaasMaster branch of private repo.  We may need a different label for this scenario.

**Comment by konrad-jamrozik** [comment]
@ladonnaq @JeffreyRichter @ronniegeraghty should we **require** for the new ARM Resource Provider SDK review be **completed** by the SDK architects before we merge such a PR, or just that an issue to get such an SDK architect review is **opened** ?

**Comment by JeffreyRichter** [comment]
This doesn't feel right to me: new-rp-ns-sdk.
RP namespaces are for service-side concepts for control-plane APIs (not data-plane APIs).
SDK is client-side and some (not all) SDK languages have a namespace which has nothing to do with an RP namespace. And, SDK namespaces apply to BOTH control-plan & data-plane.

**Comment by ronniegeraghty** [comment]
If the mgmt plane namespaces being confirmed as something different from what was originally proposed, and that causes the partner team to have to open a new PR to update their Spec, then maybe it is worth waiting until the namespace review is completed. Or does it not matter. Will making updates to the library names not trigger a new spec review PR?

**Comment by JeffreyRichter** [comment]
@ronniegeraghty : I think you're talking about the SDK package "namespace" review you send mails about periodically. If so, these agreed-to "namespaces" which are per-language do NOT belong in the specs repo - they are SDK-specific and nothing to do with the specs.

**Comment by ladonnaq** [comment]
> I just asked @konrad-jamrozik to remove this new approval label from the spec's PR process which will unblock spec PRs. This review feels like it belongs in the release planner flow when folks request a mgmt sdk release. I've brought it up to @josefree @lirenhe @praveenkuttappan in an email related to that process.

Later this semester we will be integrating the management plane SDK package namespace review and approval process into the Release Planner and API View. At this time the process is manual. However, even though it is a manual process, the "Generate SDK step" in the release plan cannot be completed until the SDK package namespace review and approval is complete.  In this step there is a drop-down box to select the SDK package namespace. The new SDK package namespace will not show up in the drop-down box until the SDKs has been generated. Therefore, a user cannot proceed to the "Release SDK step" if the SDK package namespaces have not been approved.  Also, no one can request management plane SDKs without a release plan, so I think that we have the appropriate guardrails in place to ensure that all new initial SDKs will have approved SDK package namespaces before the SDKs are requested for release.  I like the idea of raising requirements as early as possible, but the current implementation in the PR does not take into consideration the impacts across the engagement experience model, what is plan this semester, and does not cover all of the scenarios.  For example, a new service launching their first product may not complete branding/naming until after Private Preview. The current implementation would block the merge of the preview API for private preview (which is usually merged into the RPSaaSMaster branch of the API Specs private repo).  There is also no APEX launch readiness requirement to release SDKs until Public Preview, so we should not block Private Preview by requiring this task to be complete.

**Comment by konrad-jamrozik** [comment]
Thank you everyone for your input!

- [Pull Request 555655](https://devdiv.visualstudio.com/DevDiv/_git/openapi-alps/pullrequest/555655): Remove the `new-rp-namespace[-Approved]` required label rule.

@weshaggard FYI

**Comment by konrad-jamrozik** [comment]
Deployed. The `new-rp-namespace` still applies in the automation, it just doesn't produce any comment nor block anything. The label `new-rp-namespace-Approved` is present only on two already merge PRs, so I am leaving it there for reference.
